python/pyspark/pandas/supported_api_gen.py

```diff
@@ -38,7 +38,7 @@
 MAX_MISSING_PARAMS_SIZE = 5
 COMMON_PARAMETER_SET = {"kwargs", "args", "cls"}
 MODULE_GROUP_MATCH = [(pd, ps), (pdw, psw), (pdg, psg)]
-PANDAS_LATEST_VERSION = "2.2.0"
+PANDAS_LATEST_VERSION = "2.2.1"
```

Comment:
This issue predates this PR, but I note that this version requirement is more strict than our dev requirement: https://github.com/apache/spark/blob/e46d7d1c60ab82a734bb5c40229d5e72a47dd932/dev/requirements.txt#L8

One consequence of this is that trying to build the Python API docs locally can fail as follows:

```python
.../spark/python/pyspark/pandas/supported_api_gen.py:115:
  UserWarning: Warning: pandas 2.2.1 is required; your version is 2.1.4
```

We should perhaps align the dev requirement with this Pandas version requirement here.

Reply:
There are multiple ways to get an older version. Since the requirement only asks for `pandas>=1.4.4`, then setting up the virtual environment months ago will install a version that remains valid even if you rerun `pip install -r ...` today. Another way is, yes, if a different requirement specifically prevents 2.2.1 from being installed.

The way we manage Python dependencies leaves a lot to be desired. I tried to fix it in the past (e.g. #27928) but failed to get enough committer support. I am inclined to try again, because I see a constant stream of commits just trying to keep the Python build working, and I think there should be a way to make this easier for everyone.

Reply:
yes, but what about dev-container then? 
Dev container have been open sourced some months ago [Development Containers](https://containers.dev/ )

Reply:
I'm not familiar with Development Containers, but yes, there are probably many ways we can improve the situation.

What I advocated in #27928, and what I still believe is the best option for us today (with some tweaks to my original proposal), is to adopt pip-tools. That's because it's a very conservative approach that builds on our existing use of pip, and lets us focus on the technology-agnostic problem of separating Spark's direct dependencies from our build environment dependencies.